Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp

```diff
@@ -997,10 +997,14 @@ void CachedResourceStreamingClient::dataReceived(PlatformMediaResource&, const c
         GST_TRACE_OBJECT(src, "Pushing buffer of size %" G_GSIZE_FORMAT " bytes", gst_buffer_get_size(subBuffer));
         GstFlowReturn ret = gst_app_src_push_buffer(priv->appsrc, subBuffer);
 
-       if (UNLIKELY(ret != GST_FLOW_OK && ret != GST_FLOW_EOS && ret != GST_FLOW_FLUSHING)) {
+        if (UNLIKELY(ret != GST_FLOW_OK && ret != GST_FLOW_EOS && ret != GST_FLOW_FLUSHING)) {
+            gst_buffer_unref(subBuffer);
```

Comment:
This line asks for a `GRefPtr<GstBuffer>`  with a `leakRef()` in the `gst_app_src_push_buffer`, doesn't it?

Reply:
My feelings are the lifetime of this variable is very clear, and wrapping it in a smart pointer would cause more ugliness for little conceptual gain. If your feelings are stronger than mine, I'm happy to change it :)